gflownet/envs/crystals/clattice_parameters.py

```diff
@@ -0,0 +1,242 @@
+"""
+Classes to represent continuous lattice parameters environments.
+"""
+from typing import List, Optional, Tuple
+
+import numpy as np
+import torch
+from torch import Tensor
+from torchtyping import TensorType
+
+from gflownet.envs.crystals.lattice_parameters import LatticeParameters
+from gflownet.envs.cube import ContinuousCube
+from gflownet.utils.common import copy
+from gflownet.utils.crystals.constants import (
+    CUBIC,
+    HEXAGONAL,
+    LATTICE_SYSTEMS,
+    MONOCLINIC,
+    ORTHORHOMBIC,
+    RHOMBOHEDRAL,
+    TETRAGONAL,
+    TRICLINIC,
+)
+
+
+# TODO: figure out a way to inherit the (discrete) LatticeParameters env or create a
+# common class for both discrete and continous with the common methods.
+class CLatticeParameters(ContinuousCube):
+    """
+    Continuous lattice parameters environment for crystal structures generation.
+
+    Models lattice parameters (three edge lengths and three angles describing unit
+    cell) with the constraints given by the provided lattice system (see
+    https://en.wikipedia.org/wiki/Bravais_lattice). This is implemented by inheriting
+    from the (continuous) cube environment, creating a mapping between cell position
+    and edge length or angle, and imposing lattice system constraints on their values.
+
+    Similar to the Cube environment, the values are initialized with zeros
+    (or target angles, if they are predetermined by the lattice system), and are
+    incremented by sampling from a (mixture of) Beta distribution(s).
+
+    The values of the state will remain in the default [0, 1] range of the Cube, but
+    they are mapped to [min_length, max_length] in the case of the lengths and
+    [min_angle, max_angle] in the case of the angles.
+    """
+
+    def __init__(
+        self,
+        lattice_system: str,
+        min_length: float = 1.0,
+        max_length: float = 5.0,
+        min_angle: float = 30.0,
+        max_angle: float = 150.0,
+        **kwargs,
+    ):
+        """
+        Args
+        ----
+        lattice_system : str
+            One of the seven lattice systems.
+
+        min_length : float
+            Minimum value of the lengths.
+
+        max_length : float
+            Maximum value of the lengths.
+
+        min_angle : float
+            Minimum value of the angles.
+
+        max_angle : float
+            Maximum value of the angles.
+        """
+        self.lengths = ("a", "b", "c")
+        self.angles = ("alpha", "beta", "gamma")
```

Comment:
This is slightly confusing variable name, perhaps `{length, angle}_parameters`? (I would actually personally do `self.{length, angle}_parameter_names` and `self.parameter_names` in the next line.)

Reply:
This? https://github.com/alexhernandezgarcia/gflownet/pull/209/commits/2e9a1b3127152d544c8ca6e407e4bb77ab46f639